include/zephyr/sys/util.h

```diff
@@ -202,6 +202,23 @@ extern "C" {
 		(POINTER_TO_UINT(ptr) - POINTER_TO_UINT(array)) / sizeof((array)[0]);              \
 	})
 
+/**
+ * @brief Iterate over members of an array using an index variable
+ *
+ * @param array the array in question
+ * @param idx name of array index variable
+ */
+#define ARRAY_FOR_EACH(array, idx) for (size_t idx = 0; (idx) < ARRAY_SIZE(array); ++(idx))
```

Comment:
Hmm, why `++(idx)` instead of more typical `(idx)++`, don't we miss now one value from the loop?

Reply:
Hehe.. all values are accounted for.

It's mainly to be more specific about  avoid the possibility of a compiler generating less efficient code. gcc and llvm are usually good about optimizing-out any temporaries if post-incrementing is used and no assignment is done, but some other compilers might not do that as well. With pre-incrementing, there is no temporary.

E.g. `a = b++` and `a = ++b` vs `b++` and `++b`

Reply:
Indeed :disappointed: I wonder what I was thinking here...

Reply:
This is already in main actually, so I should rebase..